src/Chainweb/Pact/TransactionExec.hs

```diff
@@ -467,16 +502,19 @@ buyGas env cmd (Miner mid mks) supply mcache = do
 --
 redeemGas
     :: CommandEnv p
-    -> Command (Payload PublicMeta ParsedCode)
+    -> Command PayloadWithText
     -> CommandResult a -- ^ result from the user command payload
     -> GasId           -- ^ result of the buy-gas continuation
     -> [TxLog Value]   -- ^ previous txlogs
     -> ModuleCache
     -> IO (T2 (CommandResult [TxLog Value]) ModuleCache)
-redeemGas env cmd cmdResult gid prevLogs mcache = do
-    let fee       = gasFeeOf (_crGas cmdResult) (gasPriceOf cmd)
-        rk        = cmdToRequestKey cmd
-        initState = set (evalRefs . rsLoadedModules) mcache
+redeemGas env cmdIn cmdResult gid prevLogs mcache = do
+    let cmd        = payloadObj <$> cmdIn
+        initialGas = initialGasOf (payloadWithText cmdIn)
```

Comment:
this function is only called from within `applyCmd`, it should remain largely unchanged and definitely should not be recomputing `initialGasOf`, pass it in from `applyCmd`

Reply:
see latest commit